Shopify/cli
# Make inquirer default for inputs

### WHY are these changes introduced?

Fixes https://github.com/Shopify/shopify-cli-planning/issues/298

### WHAT is this pull request doing?
- Refactor rendering prompts library selection and execution trying to add a higher abstraction level
- Inquirer prompts are used by default. 
- Enquirer integration remains available using the env variable SHOPIFY_USE_ENQUIRER
- There are no differences in the behaviour of `select` component
- There are no differences in the behaviour of `password` component
- There are slight differences in the behaviour `input` and `autocomplete`. Please click on the sections below to review them
<details>
  <summary>Old Input vs. New Input</summary>
<br/>
Old Input
<br/>
<img src="https://user-images.githubusercontent.com/105213827/182864289-7e6a78f9-b6d6-4e10-9340-fae6361a0d6f.gif" width="750"/>
<br/>
<br/>
<br/>
New Input
<br/>
<img src="https://user-images.githubusercontent.com/105213827/182864368-03687bed-154e-4d76-9c3b-b1f73aa84d48.gif" width="750"/>
<br/>
<br/>
<br/>
Diferences:
<br>
<ul>
  <li>Cursor color</li>
  <li>Underline style, thinner and separated from the input characters</li>
</ul>
</details>

<details>
  <summary>Old autocomplete vs. New autocomplete</summary>
<br/>
Old autocomplete
<br/>
<img src="https://user-images.githubusercontent.com/105213827/183048617-086965b4-6db4-4c31-8310-ca70f4041ffc.gif" width="450"/>
<br/>
<br/>
<br/>
New autocomplete
<br/>
<img src="https://user-images.githubusercontent.com/105213827/183048687-23f23c3d-883d-4af2-97e8-d1bbb54f93dc.gif" width="400"/>
<br/>
<br/>
<br/>
Diferences:
<br>
<ul>
  <li>Paginated choices (fixed to 10)</li>
</ul>
</details>




### How to test your changes?
- Run the command `SHOPIFY_USE_INQUIRER=1 yarn shopify app dev --reset`
  - First prompt is using new `select` component
    - Select `Yes, create it as a new app` to check the `input` component
    - Select `No, connect it to an existing app` to check the `autocomplete` component. Some app should be created first
